main.go

```diff
@@ -25,17 +25,19 @@ import (
 )
 
 var (
-	showVersion    = flag.Bool("version", false, "Show version")
-	debug          = flag.Bool("debug", false, "Debug")
-	listen         = flag.String("listen", ":53", "Address to listen to (TCP and UDP)")
-	listenReload   = flag.String("listenReload", "127.0.0.1:8113", "Address to listen to for reload requests (TCP)")
-	answersFile    = flag.String("answers", "./answers.yaml", "File containing the answers to respond with")
-	defaultTtl     = flag.Uint("ttl", 600, "TTL for answers")
-	ndots          = flag.Uint("ndots", 0, "Queries with more than this number of dots will not use search paths")
-	cacheCapacity  = flag.Uint("cache-capacity", 1000, "Cache capacity")
-	logFile        = flag.String("log", "", "Log file")
-	pidFile        = flag.String("pid-file", "", "PID to write to")
-	metadataServer = flag.String("metadata-server", "", "Metadata server url")
+	showVersion              = flag.Bool("version", false, "Show version")
+	debug                    = flag.Bool("debug", false, "Debug")
+	listen                   = flag.String("listen", ":53", "Address to listen to (TCP and UDP)")
+	listenReload             = flag.String("listenReload", "127.0.0.1:8113", "Address to listen to for reload requests (TCP)")
+	answersFile              = flag.String("answers", "./answers.yaml", "File containing the answers to respond with")
+	defaultTtl               = flag.Uint("ttl", 600, "TTL for answers")
+	ndots                    = flag.Uint("ndots", 0, "Queries with more than this number of dots will not use search paths")
+	cacheCapacity            = flag.Uint("cache-capacity", 1000, "Cache capacity")
+	logFile                  = flag.String("log", "", "Log file")
+	pidFile                  = flag.String("pid-file", "", "PID to write to")
+	metadataServer           = flag.String("metadata-server", "", "Metadata server url")
+	metadataAdvertiseAddress = flag.String("metadata-advertise-address", "169.254.169.250", "Metadata advertise address (static A record)")
```

Comment:
The advertiseAddress (previously a constant) is the A record served for resolving `rancher-metdata`. It is the address that external consumers would use to connect to metadata. Previously, this was the link-local address `169.254.169.250`, but we wanted to use a routable IP.

The `metadata-server` is the address which the DNS server uses to connect to metadata. Since metadata and dns share the same network namespace, we can specify `localhost`. 

The link-local (or routable) IP could be used for both, but doing so might be less performant.